images/bootstrap/pkg/service/cm_lock.go

```diff
@@ -0,0 +1,82 @@
+package service
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ConfigMapLock struct {
+	kubernetes.Interface
+}
+
+func (c *ConfigMapLock) AcquireLock() (*v1.ConfigMap, error) {
+	cfg, err := c.CoreV1().ConfigMaps(namespace).Get(configmap, meta_v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Warnf("bootstrap: configmap cassandra-topology configmap %s cannot be found...", configmap)
+		return nil, err
+	}
+	if cfg.Annotations == nil {
+		cfg.Annotations = make(map[string]string)
+	}
+	cfg.Name = configmap
+	if cfg.Annotations[ANNOTATION_LOCK] == "" || cfg.Annotations[ANNOTATION_LOCK] == pod {
+		cfg.Annotations[ANNOTATION_LOCK] = pod
+		log.Infof("bootstrap: acquiring annotationLock of configmap %s/%s", namespace, configmap)
+		return c.CoreV1().ConfigMaps(namespace).Update(cfg)
+	}
+	return nil, fmt.Errorf("cannot acquire lock for %s. pod %s has the lock", configmap, cfg.Annotations[ANNOTATION_LOCK])
+}
+
+func (c *ConfigMapLock) HasLock() (*v1.ConfigMap, error) {
+	cfg, err := c.CoreV1().ConfigMaps(namespace).Get(configmap, meta_v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Warnf("bootstrap: configmap cassandra-topology configmap %s cannot be found...", configmap)
+		return nil, err
+	}
+	if cfg.Annotations == nil {
+		return nil, fmt.Errorf("configmap %s has no annotations", configmap)
+	}
```

Comment:
Is this an error that needs to be specifically handled?

I'd probably just maek the next line `if cfg.Annotations != nil && cfg.Annotations[ANNOTATION_LOCK] == pod`

But it may be an actual unexpected state